client/deposits/instant-deposits/index.js

```diff
@@ -6,72 +6,37 @@
 import { Button } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { useState } from '@wordpress/element';
-import apiFetch from '@wordpress/api-fetch';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
 import InstantDepositModal from './modal';
+import { useInstantDeposit } from 'data';
 
 const InstantDepositButton = ( {
-	// eslint-disable-next-line camelcase
-	balance: { amount, fee, net, transaction_ids },
+	balance: { amount, fee, net, transaction_ids: transactionIds },
 } ) => {
 	const [ isModalOpen, setModalOpen ] = useState( false );
-	const [ inProgress, setInProgress ] = useState( false );
-	const [ notice, setNotice ] = useState( null );
-
-	// TODO: Use wp.data
-	const submit = async () => {
-		try {
-			setInProgress( true );
-			setNotice( false );
-			await apiFetch( {
-				path: '/wc/v3/payments/deposits',
-				method: 'POST',
-				data: {
-					type: 'instant',
-					// eslint-disable-next-line camelcase
-					transaction_ids,
-				},
-			} );
-			// TODO: Success notice? Full-reload the page so the new deposit appears?
-			setNotice( { result: 'success' } );
-		} catch ( err ) {
-			setNotice( {
-				result: 'error',
-				code: err.code,
-				message: err.message,
-			} );
-		} finally {
-			setInProgress( false );
-		}
-	};
+	const { deposit, inProgress, submit } = useInstantDeposit( transactionIds );
 
 	const onClose = () => {
-		// If it has a notice, we want to refresh the page.
-		if ( notice ) {
-			document.location.reload();
-		} else {
-			setModalOpen( false );
-		}
+		setModalOpen( false );
 	};
 
 	return (
 		<>
 			<Button isDefault onClick={ () => setModalOpen( true ) }>
```

Comment:
I already saw you mentioned something about buttons styles in #1335. But I think we should track this. Because as noted [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CHANGELOG.md#deprecations-2) we need to replace `isDefault` with `isSecondary` when #1536 get merged.

Reply:
Added TODO in two spots for this.